awx/ui/client/src/templates/workflows/workflow-maker/workflow-maker.controller.js

```diff
@@ -381,88 +350,94 @@ export default ['$scope', 'WorkflowService', 'GetBasePath', 'TemplatesService',
                         return TemplatesService.disassociateWorkflowNode({
                             parentId: request.parentId,
                             nodeId: request.nodeId,
                             edge: request.edge
                         });
                     });
 
-                    let editNodePromises = editRequests.map(function(request) {
+                    let editNodePromises = editRequests.map(function (request) {
                         return TemplatesService.editWorkflowNode({
                             id: request.id,
                             data: request.data
                         });
                     });
 
-                    let deletePromises = $scope.treeData.data.deletedNodes.map(function(nodeId) {
+                    let deletePromises = $scope.treeData.data.deletedNodes.map(function (nodeId) {
                         return TemplatesService.deleteWorkflowJobTemplateNode(nodeId);
                     });
 
                     $q.all(disassociatePromises.concat(editNodePromises, deletePromises))
-                    .then(function() {
+                        .then(function () {
 
-                        let credentialPromises = credentialRequests.map(function(request) {
-                            return TemplatesService.postWorkflowNodeCredential({
-                                id: request.id,
-                                data: request.data
+                            let credentialPromises = credentialRequests.map(function (request) {
+                                return TemplatesService.postWorkflowNodeCredential({
+                                    id: request.id,
+                                    data: request.data
+                                });
                             });
-                        });
 
-                        let associatePromises = associateRequests.map(function(request) {
-                            return TemplatesService.associateWorkflowNode({
-                                parentId: request.parentId,
-                                nodeId: request.nodeId,
-                                edge: request.edge
+                            let associatePromises = associateRequests.map(function (request) {
+                                return TemplatesService.associateWorkflowNode({
+                                    parentId: request.parentId,
+                                    nodeId: request.nodeId,
+                                    edge: request.edge
+                                });
                             });
-                        });
 
-                        $q.all(associatePromises.concat(credentialPromises))
-                        .then(function() {
-                            $scope.closeDialog();
-                        }).catch(({data, status}) => {
+                            $q.all(associatePromises.concat(credentialPromises))
```

Comment:
I think you can avoid having to duplicate the `catch` block in the inner promise by adding a `return` statement here.